src/reducers/versions.spec.tsx

```diff
@@ -268,4 +290,111 @@ describe(__filename, () => {
       expect(dispatch).not.toHaveBeenCalled();
     });
   });
+
+  describe('fetchVersionFile', () => {
+    it('calls getVersion', async () => {
+      const version = fakeVersion;
+      const _getVersion = jest.fn().mockReturnValue(Promise.resolve(version));
+
+      const addonId = 123;
+      const versionId = version.id;
+      const path = 'some/path.js';
+
+      const { store, thunk } = thunkTester({
+        createThunk: () =>
+          fetchVersionFile({ _getVersion, addonId, versionId, path }),
+      });
+
+      await thunk();
+
+      expect(_getVersion).toHaveBeenCalledWith({
+        addonId,
+        apiState: store.getState().api,
+        versionId,
+        path,
+      });
+    });
+
+    it('dispatches updateSelectedPath', async () => {
+      const version = fakeVersion;
+      const _getVersion = jest.fn().mockReturnValue(Promise.resolve(version));
+
+      const addonId = 123;
+      const versionId = version.id;
+      const path = 'some/path.js';
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () =>
+          fetchVersionFile({
+            _getVersion,
+            addonId,
+            versionId,
+            path,
+          }),
+      });
+
+      await thunk();
+
+      expect(dispatch).toHaveBeenCalledWith(
+        actions.updateSelectedPath({ selectedPath: path, versionId }),
+      );
+    });
+
+    it('dispatches loadVersionFile when API response is successful', async () => {
+      const version = fakeVersion;
+      const _getVersion = jest.fn().mockReturnValue(Promise.resolve(version));
+
+      const addonId = 123;
+      const versionId = version.id;
+      const path = 'some/path.js';
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () =>
+          fetchVersionFile({
+            _getVersion,
+            addonId,
+            versionId,
+            path,
+          }),
+      });
+
+      await thunk();
+
+      expect(dispatch).toHaveBeenCalledWith(
+        actions.loadVersionFile({
+          path,
+          version,
+        }),
+      );
+    });
+
+    it('logs an error when API response is not successful', async () => {
```

Comment:
Perhaps we should also test that `loadVersionFile` is not dispatched when there is an error, either by adding an assertion into this test case, or creating an additional test case?

Reply:
I added an assertion.